ra2ce/analysis/losses/resilience_curves/resilience_curves.py

```diff
@@ -0,0 +1,64 @@
+"""
+                    GNU GENERAL PUBLIC LICENSE
+                      Version 3, 29 June 2007
+    Risk Assessment and Adaptation for Critical Infrastructure (RA2CE).
+    Copyright (C) 2023 Stichting Deltares
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from dataclasses import dataclass, field
+
+import numpy as np
+
+from ra2ce.network.network_config_data.enums.road_type_enum import RoadTypeEnum
+
+
+@dataclass(kw_only=True)
+class ResilienceCurves:
```

Comment:
Missing docstrings for all 'public' methods. I am specially curious if the `get_disruption` method is 'safe' against operation of arrays with different dimensions.

Reply:
Some of the docstrings are added in #504 already. I'll add the others there as well to avoid merge conflicts.
What dimensions do you mean? For a certain `link_type`/`hazard_range` the dimensions for `duration_steps` and `functionality_loss_ratio` should be equal. This is enforced in `ResilienceCurvesReader`.

Reply:
Add check + documentation